pkg/kubelet/pod_workers.go

```diff
@@ -936,7 +1031,24 @@ func (p *podWorkers) completeTerminated(pod *v1.Pod) {
 	}
 	delete(p.podUpdates, pod.UID)
 	delete(p.lastUndeliveredWorkUpdate, pod.UID)
-	delete(p.terminatingStaticPodFullnames, kubecontainer.GetPodFullName(pod))
+	if fullname, exists := p.podFullnames[pod.UID]; exists {
```

Comment:
Is this chunk and the following the same?

Reply:
Why are you doing the logic to keep waitingToStartStaticPods in `complete*()` vs inside of `allowNextStaticPod()`?  Is there a reason to have it here?